### PR TITLE
[PUB-1919]Add Preview Save Note functionality

### PR DESCRIPTION
**Pull request by anafilipadealmeida**
## Description
 - Add Preview Save Note functionality in the Queue
- Add Preview Save Note functionality in the Composer
- Change numberOfBarsFilled in the ProgressBar (order started at 0, now it starts at 1)
- Create utils package for Preview: updateNote and commonProps. 

## Context & Notes
[PUB-1919](https://buffer.atlassian.net/browse/PUB-1919)

## Checklist

-   [x] My code follows the code style and guidelines of this project. 
-   [x] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [x] I have tested different user stories. 
-   [x] I have identified similar or related features and tested they are still working.
-   [x] I have performed a self-review of my own code.
-   [x] I have tested my changes/additions in the latest Chrome, Firefox, and Safari.
-   [x] I have commented my code, particularly in hard-to-understand areas.

#### Staging Deployment

To visit the URL of the staging deployment, please click "Show all checks" at the bottom of this PR and click "details" next to `bufferbotbrains/cicd-buffer-publish-legacy`

### CODE REVIEW

File: packages/shared-components/Story/index.jsx [hunk 1]
```diff
@@ -28,7 +31,7 @@ const Story = ({
         creatorName={storyDetails.creatorName}
         avatarUrl={storyDetails.avatarUrl}
         createdAt={storyDetails.createdAt}
-        onPreviewClick={() => onPreviewClick(storyDetails.stories)}
+        onPreviewClick={() => onPreviewClick(storyDetails.stories, profileId, id, scheduledAt)}
```
Comment (albennett) [inline packages/shared-components/Story/index.jsx]: this isn't a blocker, but I normally try to pass in an object if passing in more than one argument. I find it safer in preventing arguments getting passed out of order.
https://www.notion.so/buffer/Frontend-Style-Guide-158952c1e97c4eaa903ab4fce233107f#f9e7a824197e491393631365628363da

File: packages/story-preview/reducer.js [hunk 2]
```diff
@@ -1,14 +1,28 @@
...
 
 export const initialState = {
   user: {},
   stories: [],
+  profileId: null,
+  storyGroupId: null,
+  scheduledAt: null,
+};
+
+const updateStoryNote = ({ stories = [], order, note }) => {
```
Comment (albennett) [inline packages/story-preview/reducer.js]: I wonder if we could add this method to a utils so it could be used in both the reducer & middleware.

Comment (anafilipadealmeida) [reply packages/story-preview/reducer.js]: I was doing it in another PR, but I've done it now and added proper testing to the utils 🙂

File: packages/shared-components/Story/index.jsx (hunk 1 above)
Comment (anafilipadealmeida) [reply packages/shared-components/Story/index.jsx]: You are absolutely right @albennett, this was a terrible way of passing properties! It's fixed 🙂

**Pull request merged**